projects/igniteui-angular/src/lib/grid/grid-filtering-ui.spec.ts

```diff
@@ -987,116 +958,101 @@ describe('IgxGrid - Filtering actions', () => {
+        fix.detectChanges();
+        tick(100);
 
-            fix.detectChanges();
+        input.nativeElement.click();
+        fix.detectChanges();
+        tick();
 
-            const firstMonth = calendar.queryAll(By.css('.igx-calendar__month'))[0];
-            firstMonth.nativeElement.click();
+        let calendar = fix.debugElement.query(By.css('igx-calendar'));
+        const monthView = calendar.queryAll(By.css('.date__el'))[0];
+        monthView.nativeElement.click();
+        fix.detectChanges();
+        tick();
 
-            return fix.whenStable();
-        }).then(() => {
-            fix.detectChanges();
+        const firstMonth = calendar.queryAll(By.css('.igx-calendar__month'))[0];
+        firstMonth.nativeElement.click();
+        fix.detectChanges();
+        tick();
 
-            const calendar = fix.debugElement.query(By.css('igx-calendar'));
-            const month = calendar.queryAll(By.css('.date__el'))[0];
+        calendar = fix.debugElement.query(By.css('igx-calendar'));
+        const month = calendar.queryAll(By.css('.date__el'))[0];
 
-            expect(month.nativeElement.textContent.trim()).toEqual('Jan');
-        });
+        expect(month.nativeElement.textContent.trim()).toEqual('Jan');
     }));
 
-    it('Should correctly select year from year view datepicker/calendar component', async(() => {
+    it('Should correctly select year from year view datepicker/calendar component', fakeAsync(() => {
         const fix = TestBed.createComponent(IgxGridFilteringComponent);
         fix.detectChanges();
         const filterIcon = fix.debugElement.queryAll(By.css(FILTER_UI_CONTAINER))[3];
         const input = filterIcon.query(By.directive(IgxInputDirective));
 
-        fix.whenStable().then(() => {
-            filterIcon.triggerEventHandler('mousedown', null);
-            fix.detectChanges();
-            filterIcon.nativeElement.click();
-            fix.detectChanges();
-            return fix.whenStable();
-        }).then(() => {
-            fix.detectChanges();
-
-            input.nativeElement.click();
-            fix.detectChanges();
+        filterIcon.triggerEventHandler('mousedown', null);
+        fix.detectChanges();
+        filterIcon.nativeElement.click();
+        fix.detectChanges();
+        tick(100);
 
-            const calendar = fix.debugElement.query(By.css('igx-calendar'));
-            const monthView = calendar.queryAll(By.css('.date__el'))[1];
-            monthView.nativeElement.click();
+        input.nativeElement.click();
+        fix.detectChanges();
+        tick();
 
-            fix.detectChanges();
+        let calendar = fix.debugElement.query(By.css('igx-calendar'));
+        const monthView = calendar.queryAll(By.css('.date__el'))[1];
+        monthView.nativeElement.click();
 
-            const firstMonth = calendar.queryAll(By.css('.igx-calendar__year'))[0];
-            firstMonth.nativeElement.click();
+        fix.detectChanges();
```

Comment:
`tick` maybe? Not sure if the following line command needs to wait for this click.